GitUI/CommandsDialogs/SettingsDialog/Pages/FormBrowseRepoSettingsPage.cs

```diff
@@ -1,78 +1,502 @@
+            }
+
+            public override StandardValuesCollection GetStandardValues(ITypeDescriptorContext context)
+            {
+                var value = (string)context.PropertyDescriptor.GetValue(context.Instance);
+
+                return new StandardValuesCollection(ResourceImagesProvider.Images.Select(x => x.Key).ToArray());
+            }
+        }
+
+        private sealed class ResourceImagesTypeEditor : UITypeEditor
+        {
+            public override UITypeEditorEditStyle GetEditStyle(ITypeDescriptorContext context)
+            {
+                return UITypeEditorEditStyle.DropDown;
+            }
+
+            public override bool GetPaintValueSupported(ITypeDescriptorContext context)
+            {
+                return true;
+            }
+
+            public override void PaintValue(PaintValueEventArgs e)
+            {
+                var imageKey = e.Value?.ToString();
+
+                if (imageKey != null)
+                {
+                    var rectangle = e.Bounds;
+
+                    rectangle.Inflate(-2, -1);
+
+                    e.Graphics.DrawImage(ResourceImagesProvider.Images[imageKey], rectangle);
+                }
+            }
+
+            public override object EditValue(ITypeDescriptorContext context, IServiceProvider provider, object value)
+            {
+                var editorService = (IWindowsFormsEditorService)provider.GetService(typeof(IWindowsFormsEditorService));
+
+                if (editorService is null)
+                {
+                    return value;
+                }
+
+                var listView = new ListView
+                {
+                    Scrollable = true,
+                    View = View.List,
+                    MultiSelect = false,
+                    BorderStyle = BorderStyle.None
+                };
+
+                var imageList = new ImageList
+                {
+                    ColorDepth = ColorDepth.Depth32Bit
+                };
+
+                listView.SmallImageList = imageList;
+                listView.LargeImageList = imageList;
+
+                foreach (var image in ResourceImagesProvider.Images)
+                {
+                    imageList.Images.Add(image.Key, image.Value);
+
+                    listView.Items.Add(image.Key, image.Key);
+                }
+
+                listView.Click += (object sender, EventArgs e) =>
+                {
+                    editorService.CloseDropDown();
+                };
+
+                editorService.DropDownControl(listView);
+
+                if (listView.SelectedItems.Count > 0)
+                {
+                    return listView.SelectedItems[0].Text;
+                }
+
+                return value;
+            }
         }
 
-        private void cboTerminal_Enter(object sender, EventArgs e)
+        [TypeConverter(typeof(PropertySorter))]
+        private sealed class ShellViewModel : IShell, INotifyPropertyChanged, ICheckable
```

Comment:
We don't realy use View-ViewModel-Model naming convention. I suppose we can, but we'll need to be consistent and rename `ScriptInfoProxy`.

Reply:
I think this is not really a proxy